entities/fraction.go

```diff
@@ -120,16 +120,32 @@ func (f *Fraction) Divide(other *Fraction) *Fraction {
 func (f *Fraction) ToSignificant(significantDigits uint, opt ...number.Option) string {
 	f.opts = number.New(number.WithGroupSeparator('\xA0'), number.WithRoundingMode(constants.RoundHalfUp))
 	f.opts.Apply(opt...)
-	f.opts.Apply(number.WithRoundingPrecision(int(significantDigits)))
 
 	d := decimal.NewFromBigInt(f.Numerator, 0).Div(decimal.NewFromBigInt(f.Denominator, 0))
+	if d.LessThan(decimal.New(1, 0)) {
+		significantDigits += countZerosAfterDecimalPoint(d.String())
+	}
+	f.opts.Apply(number.WithRoundingPrecision(int(significantDigits)))
 	if v, err := number.DecimalRound(d, f.opts); err == nil {
 		d = v
 	}
-
+	fmt.Println(d.String())
```

Comment:
could you help to remove the debug code?

Reply:
ah dammit, thought I removed everything.. anyway. It's removed now